server/channels/store/sqlstore/store.go

```diff
@@ -239,24 +241,30 @@ func New(settings model.SqlSettings, metrics einterfaces.MetricsInterface) (*Sql
 
 // SetupConnection sets up the connection to the database and pings it to make sure it's alive.
 // It also applies any database configuration settings that are required.
-func SetupConnection(connType string, dataSource string, settings *model.SqlSettings, attempts int) (*dbsql.DB, error) {
+func SetupConnection(logger mlog.LoggerIFace, connType string, dataSource string, settings *model.SqlSettings, attempts int) (*dbsql.DB, error) {
 	db, err := dbsql.Open(*settings.DriverName, dataSource)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to open SQL connection")
 	}
 
+	// At this point, we have passed sql.Open, so we deliberately ignore any errors.
+	sanitized, _ := SanitizeDataSource(*settings.DriverName, dataSource)
+
+	logger = logger.With(
+		mlog.String("database", connType),
+		mlog.String("dataSource", sanitized),
+	)
+
 	for i := 0; i < attempts; i++ {
-		// At this point, we have passed sql.Open, so we deliberately ignore any errors.
-		sanitized, _ := SanitizeDataSource(*settings.DriverName, dataSource)
-		mlog.Info("Pinging SQL", mlog.String("database", connType), mlog.String("dataSource", sanitized))
+		logger.Info("Pinging SQL")
 		ctx, cancel := context.WithTimeout(context.Background(), DBPingTimeoutSecs*time.Second)
 		defer cancel()
 		err = db.PingContext(ctx)
 		if err != nil {
 			if i == attempts-1 {
 				return nil, err
 			}
-			mlog.Error("Failed to ping DB", mlog.Err(err), mlog.Int("retrying in seconds", DBPingTimeoutSecs))
+			logger.Error("Failed to ping DB", mlog.Int("retrying in seconds", DBPingTimeoutSecs), mlog.Err(err))
 			time.Sleep(DBPingTimeoutSecs * time.Second)
 			continue
```

Comment:
The changes here should be double-checked, but I think they are fine.

Reply:
Good catch 👍